pkg/odo/cli/component/create.go

```diff
@@ -881,6 +881,101 @@ func (co *CreateOptions) Validate() (err error) {
 	return nil
 }
 
+func downloadGitProject(starterProject *devfilev1.StarterProject, starterToken, path string) error {
```

Comment:
Agreed that it's a private function, but it's large. Some comments, maybe?

Reply:
Added